hathor/p2p/sync_v2/agent.py

```diff
@@ -341,7 +326,7 @@ def run_sync_transactions(self) -> Generator[Any, Any, None]:
 
         self.log.info('run sync transactions', start=[i.hex() for i in needed_txs], end_block_hash=block.hash.hex(),
                       end_block_height=block_height)
-        yield self.start_transactions_streaming(needed_txs, block.hash)
+        yield self.start_transactions_streaming(needed_txs, block.hash, block.hash)
```

Comment:
What should we do if the deferred fails? Maybe set state to ERROR? Or just close connection?

Reply:
I agree. This code was completely removed in PR #850. So I think we should discuss the final implementation there. Is it ok for you?

Reply:
I added a try/except block and just marked streaming as false. Check it out here: https://github.com/HathorNetwork/hathor-core/pull/848/commits/c044befd04c533c13894a8d39258287fa5549e4d